packages/react-composites/src/composites/MeetingComposite/adapter/MeetingAdapter.ts

```diff
@@ -0,0 +1,131 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+/* eslint-disable @typescript-eslint/adjacent-overload-signatures */
+
+import {
+  CallAdapterCallManagement,
+  CallAdapterDeviceManagement,
+  CallIdChangedListener,
+  DisplayNameChangedListener,
+  IsMuteChangedListener,
+  IsScreenSharingOnChangedListener,
+  IsSpeakingChangedListener,
+  ParticipantJoinedListener,
+  ParticipantLeftListener
+} from '../../CallComposite';
+import {
+  ChatAdapterThreadManagement,
+  MessageReadListener,
+  MessageReceivedListener,
+  MessageSentListener
+} from '../../ChatComposite';
+import { MeetingState } from '../state/MeetingState';
+
+import type { AdapterState, AdapterDisposal, AdapterPages } from '../../common/adapters';
+
+/**
+ * Page state the Meeting composite could be in.
+ * @alpha
+ */
+export type MeetingCompositePage = 'configuration' | 'meeting' | 'error' | 'errorJoiningTeamsMeeting' | 'removed';
+
+/**
+ * Functionality for managing the current meeting.
+ * @alpha
+ */
+export interface MeetingAdapterMeetingManagement
+  extends Pick<
+      CallAdapterCallManagement,
+      | 'joinCall'
+      | 'leaveCall'
+      | 'startCamera'
+      | 'stopCamera'
+      | 'onToggleCamera'
+      | 'mute'
+      | 'unmute'
+      | 'startCall'
+      | 'startScreenShare'
+      | 'stopScreenShare'
+      | 'createStreamView'
+      | 'disposeStreamView'
+    >,
+    Pick<
+      CallAdapterDeviceManagement,
+      | 'setCamera'
+      | 'setMicrophone'
+      | 'setSpeaker'
+      | 'askDevicePermission'
+      | 'queryCameras'
+      | 'queryMicrophones'
+      | 'querySpeakers'
+    >,
+    Pick<
+      ChatAdapterThreadManagement,
+      'fetchInitialData' | 'sendMessage' | 'sendReadReceipt' | 'sendTypingIndicator' | 'loadPreviousChatMessages'
+    > {
+  removeParticipant(userId: string): Promise<void>;
+}
+
+/**
+ * Meeting events that can be subscribed to.
+ * @alpha
+ */
+export interface MeetingAdapterSubscriptions {
+  // Meeting specific subscriptions
+  on(event: 'participantsJoined', listener: ParticipantJoinedListener): void;
+  on(event: 'participantsLeft', listener: ParticipantLeftListener): void;
+  on(event: 'meetingEnded', listener: ParticipantLeftListener): void;
+  on(event: 'error', listener: (e: Error) => void): void;
+
+  off(event: 'participantsJoined', listener: ParticipantJoinedListener): void;
+  off(event: 'participantsLeft', listener: ParticipantLeftListener): void;
+  off(event: 'meetingEnded', listener: ParticipantLeftListener): void;
+  off(event: 'error', listener: (e: Error) => void): void;
+
+  // Call subscriptions
```

Comment:
Is it possible to reuse interface from callSubscriber and chatSubscriber?

Reply:
I would love to be able to but can't see a way from how we do the `on` interface. One option is to change `on('string', ...)` to be `onString(...)`